src/main/java/uk/gov/ons/ctp/response/sample/service/SampleService.java

```diff
@@ -73,14 +79,18 @@ SampleSummary processSampleSummary(SurveyBase surveySampleObject, List<? extends
    */
   PartyDTO sendToPartyService(PartyCreationRequestDTO party) throws Exception;
 
+    Optional<SampleSummary> failSampleSummary(SampleSummary sampleSummary, String message);
+
+  Optional<SampleSummary> failSampleSummary(SampleSummary sampleSummary, Exception exception);
+
   /**
    * Ingest Survey Sample
    *
    * @param file Multipart File of SurveySample to be used
    * @param type Type of Survey to be used
    * @throws Exception exception thrown
    */
-  SampleSummary ingest(MultipartFile file, String type) throws Exception;
+  Pair<SampleSummary, Future<Optional<SampleSummary>>> ingest(MultipartFile file, String type) throws Exception;
```

Comment:
Can we be more specific on the type of Exception thrown? If it won't always be CTPException then fair enough but it feels like it will be.

Reply:
This isn't my code - I thought exactly the same so made the change then discovered deep down in the bowels a 3rd party library that throws base Exception.  We could wrap that up I suppose but Exception is a valid expression of the exceptions that are thrown.

Reply:
Nope, not at all worth doing. Disregard!

Reply:
Why are we returning a Pair<SampleSummary, Future<Optional<SampleSummary>>>? Why not just return the SampleSummary?

Reply:
e.g. https://github.com/ONSdigital/rm-sample-service/blob/3155782b1f407f597138ecac918a56296e761a4b/src/test/java/uk/gov/ons/ctp/response/sample/service/impl/SampleServiceImplTest.java#L323

Reply:
In this scenario it feels like it's being done just for the test potentially then? I understand the fact we could use the  futures to wait for the result so we don't have to poll the database but we don't have that requirement at the moment.

Why not keep SampleServiceImpl as synchronous and put the ExecutorService in the SampleEndpoint making SampleServiceImpl easier to test and the SampleEndpoint will still be unit testable as we'll just be checking the http response

Reply:
I know you and I disagree philosophically about how exactly a feature should meet the requirements but I am really not happy brain damaging a feature so it does not step one inch over the current requirements.  Providing there are tests to cover it I am inclined to leave it as is

Reply:
My concern is with the complexity of the API and it can be simplified. I'd prefer it was simple so it's more maintainable going forward

Reply:
Perfection is the enemy of good enough :-P